sound/soc/sof/sof-audio.h

```diff
@@ -376,6 +376,7 @@ struct snd_sof_widget {
 	 * from D3.
 	 */
 	bool dynamic_pipeline_widget;
+	u32 output_cnt;	/* the output number of the sof widget */
```

Comment:
output number? or number of output pins? Isnt @aiChaoSONG also adding something similar?

Reply:
I have added the num_source_pins token, which is used to tell how many outputs a widget can support. but the output_cnt has a totally different meaning, it means how many output formats in the initialization blob to be set for a widget. btw， copier has one.

Reply:
The output format in initialization blob is used to setup pin 0, other pins don't use it. for example, copier, format for pin 1~ pin3 is set by set sink format ipc(largeconfigset) individually.

Reply:
The comment is still not explanatory @libinyang output number? number of output ports? number of sink pins? there are too many things that seem to indicate the same thing

Reply:
output number (now it's output format number) means which output format it is using in the init instance payload. A module may have several output ports but the init instance payload may have only 1 output format data. For example, copier has several output ports. But the nit instance payload only have one output fmt when send the init instance command.